### PR TITLE
DB-7895 Query predicate rewrite with DNF to CNF conversion

### PR DESCRIPTION
**Pull request by changli6**

### CODE REVIEW

**Comment by changli6** [comment]
jenkins please test branch @cdh5.12.2

File: db-engine/src/main/java/com/splicemachine/db/impl/ast/RepeatedPredicateVisitor.java [hunk 1]
```diff
@@ -164,24 +164,34 @@ public Visitable defaultVisit(Visitable node) throws StandardException {
...
-            if (n<=1)
-                 return updatedNode;
+            if (n <= 1)
+                return updatedNode;
 
             // calculate the number of clauses in the predicate
             Map<ValueNode, Integer> m = nodesWithMultipleOccurrences(newNode, n);
 
-            for(Map.Entry<ValueNode,Integer> me : m.entrySet()) {
+            for (Map.Entry<ValueNode, Integer> me : m.entrySet()) {
```
Comment (jaceklaskowski) [inline db-engine/src/main/java/com/splicemachine/db/impl/ast/RepeatedPredicateVisitor.java]: nit: what about `me : m` being `me: m` instead?

**Comment by jaceklaskowski** [comment]
I'd also test against `cdh5.14.0` which seems to become the default profile [soon](https://github.com/splicemachine/spliceengine/pull/2729).

**Comment by spliceci** [comment]
TEST FAILED

**Comment by changli6** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**